docker/compose/dev/build/modify-configs.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+docker_compose_file="../docker-compose.yml"
+
+# Extract DEMO_HOST_ADDRESS and otherwise default to localhost
+demo_host_address=$(awk '/ssi-agent:/{flag=1; next} /environment:/{flag=2; next} flag==2 && /DEMO_HOST_ADDRESS/{split($2,a,"="); print a[2]; flag=0}' "$docker_compose_file" | tr -d '"' | tr -d ' ')
+if [ -z "$demo_host_address" ]; then
+    demo_host_address="localhost"
+fi
+
+# Check if demo_host_address contains 'http://' or 'https://' if not, prepend it
+if [[ ! $demo_host_address =~ ^http ]]; then
+    # If not, prepend 'http://'
+    demo_host_address="http://${demo_host_address}"
+fi
+
+# Extract ENVIRONMENT name and default to sphereon if not found
+environment_name=$(grep "^REACT_APP_DEFAULT_ECOSYSTEM=" ./.env.oid4vci-demo-frontend | cut -d '=' -f2)
```

Comment:
The regex is not broad enough. This will fail when someone has `KEY = value` instead of `KEY=value`, so either allow optional whitespace after the key, or remove the = altogether, since you are already cutting it at the end anyway

It will not pickup on commentend values as it looks at the beginning of the line, which means the value has to be given for it to be valid and thus always needs to include a `=`

Reply:
changed the regex